File: dynabuffers-java/src/main/kotlin/dynabuffers/ast/datatype/IntType.kt
package dynabuffers.ast.datatype

import dynabuffers.api.IRegistry
import dynabuffers.api.ISerializable
import dynabuffers.api.IType
import java.nio.ByteBuffer

class IntType : IType, ISerializable {

    override fun size(value: Any?, registry: IRegistry) = 4

    override fun serialize(value: Any?, buffer: ByteBuffer, registry: IRegistry) {
        buffer.putInt(int(value))
    }

    override fun deserialize(buffer: ByteBuffer, registry: IRegistry) = buffer.int

    private fun int(obj: Any?) = obj.toString().toInt()

}

File: dynabuffers-python/dynabuffers/ast/datatype/IntType.py
from dynabuffers.api.ISerializable import ISerializable


class IntType(ISerializable):

    def size(self, value, registry):
        return 4

    def serialize(self, value, buffer, registry):
        buffer.putInt(value if isinstance(value, int) else int(value))

    def deserialize(self, buffer, registry):
        return buffer.getInt()
